deebot_client/commands/json/ota.py

```diff
@@ -0,0 +1,51 @@
+"""Ota command module."""
+from __future__ import annotations
+
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any
+
+from deebot_client.command import InitParam
+from deebot_client.events import OtaEvent
+from deebot_client.message import HandlingResult
+
+from .common import JsonGetCommand, JsonSetCommand
+
+if TYPE_CHECKING:
+    from deebot_client.event_bus import EventBus
+
+
+class GetOta(JsonGetCommand):
+    """Get ota command."""
+
+    name = "getOta"
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        support_auto = data.get("supportAuto")
+        ota_event = OtaEvent(
+            enable=bool(data.get("autoSwitch", True)),
+            support_auto=None if support_auto is None else bool(support_auto),
```

Comment:
> Does there exists also a onOta message/event?

no, i saw only get and set

> Older bots must be updated manually

when the process for ota would be:
- on first init **get** will trigger and will informe with all possible values
- when **set** is used only `auto_enabled` is returned

with current version it would be:
```sh
# on init get
OtaEvent(enable=True, support_auto=True, version='1.7.2', status='idle', progress=0)

# when switch with set
OtaEvent(enable=False, support_auto=None, version=None, status=None, progress=None)
OtaEvent(enable=True, support_auto=None, version=None, status=None, progress=None)
OtaEvent(enable=False, support_auto=None, version=None, status=None, progress=None)
OtaEvent(enable=True, support_auto=None, version=None, status=None, progress=None)
```

with your change it would be:
```sh
# on init get
OtaEvent(enable=True, support_auto=True, version='1.7.2', status='idle', progress=0)

# when switch with set
OtaEvent(enable=False, support_auto=False, version=None, status=None, progress=None)
OtaEvent(enable=True, support_auto=False, version=None, status=None, progress=None)
OtaEvent(enable=False, support_auto=False, version=None, status=None, progress=None)
OtaEvent(enable=True, support_auto=False, version=None, status=None, progress=None)
```

when this information would be used, `support_auto` would be set with a wrong state as `False`, thats why i checked if `support_auto` is provided, and if not set to `None` to identify no change provided.

Reply:
This your concern solved with the latest implementation?

Reply:
yes, looks good so far with the latest update